hudi-common/src/main/java/org/apache/hudi/common/table/view/FileSystemViewManager.java

```diff
@@ -162,29 +162,28 @@ private static SpillableMapBasedFileSystemView createSpillableMapBasedFileSystem
    */
   private static HoodieTableFileSystemView createInMemoryFileSystemView(HoodieMetadataConfig metadataConfig, FileSystemViewStorageConfig viewConf,
                                                                         HoodieTableMetaClient metaClient, SerializableSupplier<HoodieTableMetadata> metadataSupplier) {
-    LOG.info("Creating InMemory based view for basePath " + metaClient.getBasePath());
+    LOG.info("Creating InMemory based view for basePath " + metaClient.getBasePathV2());
     HoodieTimeline timeline = metaClient.getActiveTimeline().filterCompletedAndCompactionInstants();
     if (metadataConfig.enabled()) {
       ValidationUtils.checkArgument(metadataSupplier != null, "Metadata supplier is null. Cannot instantiate metadata file system view");
-      return new HoodieMetadataFileSystemView(metaClient, metaClient.getActiveTimeline().filterCompletedAndCompactionInstants(),
-          metadataSupplier.get());
+      return new HoodieMetadataFileSystemView(metaClient, timeline, metadataSupplier.get());
     }
     if (metaClient.getMetaserverConfig().isMetaserverEnabled()) {
       return (HoodieTableFileSystemView) ReflectionUtils.loadClass(HOODIE_METASERVER_FILE_SYSTEM_VIEW_CLASS,
           new Class<?>[] {HoodieTableMetaClient.class, HoodieTimeline.class, HoodieMetaserverConfig.class},
-          metaClient, metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants(), metaClient.getMetaserverConfig());
+          metaClient, timeline, metaClient.getMetaserverConfig());
```

Comment:
This line exclude the compaction instants before, does it have regression?

Reply:
I review the Git commit history and find when instantiating HoodieMetadataFileSystemView, compaction instants were initially excluded. Later, in [HUDI-3692](https://issues.apache.org/jira/browse/HUDI-3692), this issue was fixed by using the timeline that includes compaction instants.
So I think when initializing HoodieMetaserverFileSystemView, compaction instants should also be included to ensure the  correctness of asynchronous compaction.

Reply:
Yeah, I agree.